### PR TITLE
Exposed port for LoadBalancer services (application and xsite) should be configurable

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

File: pkg/controller/infinispan/resources/service/service_controller.go [hunk 1]
```diff
@@ -416,6 +427,9 @@ func computeSiteService(ispn *ispnv1.Infinispan) *corev1.Service {
 				TargetPort: intstr.IntOrString{IntVal: consts.CrossSitePort},
 			},
 		}
+		if exposeConf.Port > 0 {
```
Comment (pruivo) [inline pkg/controller/infinispan/resources/service/service_controller.go:430]: shouldn't we do this for all `expose types`? `NodePort` and `ClusterIP` has the field `Port` too

Comment (dmvolod) [reply pkg/controller/infinispan/resources/service/service_controller.go:430]: For `NodePort` it doesn't make sense as external communication is performing via `NodePort` on all hosts. For `ClusterIP` it make sense but can guess situation why customer need to change port for internal k8s cluster communication.

Comment (pruivo) [reply pkg/controller/infinispan/resources/service/service_controller.go:430]: how about the same cluster cross-site (i.e. different namespaces/cluster names)?
 isn't this `port` be used independently of the expose type?

Comment (dmvolod) [reply pkg/controller/infinispan/resources/service/service_controller.go:430]: Yes, but for the same cluster service name is unique and port doesn't matter in this case as we are operating with service:port combination

File: test/e2e/xsite/xsite_test.go [hunk 2]
```diff
@@ -99,28 +100,32 @@ func secretSiteName(siteName string) string {
...
 }
 
 func TestCrossSiteViewOpenshiftLoadBalancer(t *testing.T) {
-	testCrossSiteView(t, true, config.SchemeTypeOpenShift, ispnv1.CrossSiteExposeTypeLoadBalancer)
+	testCrossSiteView(t, true, config.SchemeTypeOpenShift, ispnv1.CrossSiteExposeTypeLoadBalancer, 0)
 }
 
-func testCrossSiteView(t *testing.T, isMultiCluster bool, schemeType string, exposeType ispnv1.CrossSiteExposeType) {
+func TestCrossSiteViewLoadBalancerWithPort(t *testing.T) {
+	testCrossSiteView(t, true, config.SchemeTypeOpenShift, ispnv1.CrossSiteExposeTypeLoadBalancer, 1443)
```
Comment (dmvolod) [inline test/e2e/xsite/xsite_test.go:125]: @pruivo , @ryanemerson please pay attention that CI and my local machine unable to test services on privileged ports (less than 1000). Probably it's an MetalLB limitations. but need to be tested on Azure or AWS LoadBalancer

Comment (pruivo) [reply test/e2e/xsite/xsite_test.go:125]: I did a test yesterday in my laptop with port 443 and it worked fine (kind 0.11.1 + MetalLB v0.10.2)
for testing purposes, we just need the port to be different from 7900

**Review by pruivo — APPROVED** [review]

**Review by ryanemerson — APPROVED** [review]

File: test/e2e/xsite/xsite_test.go (hunk 2 above)
Comment (dmvolod) [reply test/e2e/xsite/xsite_test.go:125]: Thanks for confirmation. Will try with newer MetalLB version

**Pull request merged**